treeherder/webapp/api/logslice.py

```diff
@@ -58,9 +59,14 @@ def list(self, request, project, jm):
         if start_line >= end_line:
             return Response("``end_line`` must be larger than ``start_line``", 400)
 
-        if len(log) > 0:
+        # get only the log that matches the ``log_name``
+        logs = jm.get_log_references(job_id)
+        log = [log for log in logs if log["name"] == log_name]
+
+        if len(log):
+            log = log[0]
```

Comment:
The 'log' variable initially can actually contain multiple logs, so 'logs' might be a better name. That said, so we ever expect more than one log of each type? I wonder if in a future bug we should enforce that the job_logs_url table only has one entry of each type - or else add support for multiple logs here. Anyway I digress :-)

Seeing as we want to both filter the list of logs and also only return the first matching item, we can use Python's next() for improved perf and fewer lines. I also think it would be clearer to raise the ResourceNotFoundException early. eg:

```python
logs = jm.get_log_references(job_id)
# Get the first log found that has the desired log_name.
log = next([log for log in logs if log["name"] == log_name], None)
if not log:
    raise ResourceNotFoundException("job_artifact {0} not found".format(job_id))

try:
    url = log.get("url")
    ...
finally:
    ...
```

Or:
```python
logs = jm.get_log_references(job_id)
# Get the first log found that has the desired log_name.
try:
    log = next([log for log in logs if log["name"] == log_name])
except StopIteration:
    raise ResourceNotFoundException("job_artifact {0} not found".format(job_id))

try:
    url = log.get("url")
    ...
finally:
    ...
```